### ISSUE TITLE
Cassandra example YAMLs do not match latest version

### ISSUE THREAD

**Issue by palonsoro** [kubernetes/kubernetes#17728]
While using the YAML sample files (both cassandra.yaml and cassandra-controller.yaml) at the Cassandra example, it points to the following image:

```yaml
image: gcr.io/google_containers/cassandra:v6
```

Which seems not to contain the latest code. I noticed it when I realized that the code from `gcr.io/google_containers/cassandra:v6` tries to reach `kubernetes.default.cluster.local` and the latest code tries to reach `kubernetes.default.svc.cluster.local` to get the endpoints list.

It may be a good idea to upload a newer version of the image (if it is not already uploaded), and point the examples to it.

**Comment by jaohaohsuan** [comment]
i got same problem too.
wrong hostname https://`kubernetes.default.cluster.local`/api/v1/namespaces/default/endpoints/cassandra
```
INFO  02:54:17 Global memtable on-heap threshold is enabled at 125MB
INFO  02:54:17 Global memtable off-heap threshold is enabled at 125MB
INFO  02:54:17 Getting endpoints from https://kubernetes.default.cluster.local/api/v1/namespaces/default/endpoints/cassandra
WARN  02:54:18 Request to kubernetes apiserver failed
java.net.UnknownHostException: kubernetes.default.cluster.local
	at java.net.AbstractPlainSocketImpl.connect(AbstractPlainSocketImpl.java:178) ~[na:1.7.0_79]
	at java.net.SocksSocketImpl.connect(SocksSocketImpl.java:392) ~[na:1.7.0_79]
	at java.net.Socket.connect(Socket.java:579) ~[na:1.7.0_79]
	at sun.security.ssl.SSLSocketImpl.connect(SSLSocketImpl.java:625) ~[na:1.7.0_79]
	at sun.security.ssl.BaseSSLSocketImpl.connect(BaseSSLSocketImpl.java:160) ~[na:1.7.0_79]
	at sun.net.NetworkClient.doConnect(NetworkClient.java:180) ~[na:1.7.0_79]
	at sun.net.www.http.HttpClient.openServer(HttpClient.java:432) ~[na:1.7.0_79]
	at sun.net.www.http.HttpClient.openServer(HttpClient.java:527) ~[na:1.7.0_79]
	at sun.net.www.protocol.https.HttpsClient.<init>(HttpsClient.java:275) ~[na:1.7.0_79]
	at sun.net.www.protocol.https.HttpsClient.New(HttpsClient.java:371) ~[na:1.7.0_79]
	at sun.net.www.protocol.https.AbstractDelegateHttpsURLConnection.getNewHttpClient(AbstractDelegateHttpsURLConnection.java:191) ~[na:1.7.0_79]
	at sun.net.www.protocol.http.HttpURLConnection.plainConnect(HttpURLConnection.java:933) ~[na:1.7.0_79]
	at sun.net.www.protocol.https.AbstractDelegateHttpsURLConnection.connect(AbstractDelegateHttpsURLConnection.java:177) ~[na:1.7.0_79]
	at sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1301) ~[na:1.7.0_79]
	at sun.net.www.protocol.https.HttpsURLConnectionImpl.getInputStream(HttpsURLConnectionImpl.java:254) ~[na:1.7.0_79]
	at io.k8s.cassandra.KubernetesSeedProvider.getSeeds(KubernetesSeedProvider.java:124) ~[kubernetes-cassandra.jar:na]
	at org.apache.cassandra.config.DatabaseDescriptor.applyConfig(DatabaseDescriptor.java:654) [apache-cassandra-2.1.7.jar:2.1.7]
	at org.apache.cassandra.config.DatabaseDescriptor.<clinit>(DatabaseDescriptor.java:136) [apache-cassandra-2.1.7.jar:2.1.7]
	at org.apache.cassandra.service.CassandraDaemon.setup(CassandraDaemon.java:164) [apache-cassandra-2.1.7.jar:2.1.7]
	at org.apache.cassandra.service.CassandraDaemon.activate(CassandraDaemon.java:524) [apache-cassandra-2.1.7.jar:2.1.7]
	at org.apache.cassandra.service.CassandraDaemon.main(CassandraDaemon.java:613) [apache-cassandra-2.1.7.jar:2.1.7]
```

**Comment by jaohaohsuan** [comment]
I got the same problem.

**Comment by palonsoro** [comment]
The workaround is pretty simple: Just build locally the latest version of everything and it will work. I opened this issue so that remotely uploaded images get also updated and people do not run into problems while testing this without building.

**Comment by goltermann** [comment]
This was fixed in #21645, which now points to :v8.
